backend/src/core/services/credential.service.ts

```diff
@@ -64,6 +64,11 @@ export interface CredentialService {
     userId: number,
     announcementVersion: string | null
   ): Promise<{ announcementVersion: string }>
+
+  generateApiKey(
+    userId: number,
+    label: string
+  ): Promise<ApiKey & { key: string }>
```

Comment:
**[nit]** 
- I had to trace the code to realise that `key` is the API key in plaintext
- would prefer a more informative variable name 🤔️ maybe `apiKeyPlaintext`